platform/pddf/i2c/modules/mux/pddf_mux_module.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2019 Broadcom. All rights reserved.
+ * The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * A pddf kernel module to create I2C client for pca954x type of multiplexer
```

Comment:
Why is this driver needed when native linux pca954x driver can create devices in sysfs/devfs  as clients?

Reply:
Yes, we are also using the same PCA954x driver. However, There are some fields we have modified using the "pddf_mux_module". e.g base_bus which should be taken form device json object field 'parent_bus' which is a user-provided field. Hence we used pddf_mux_module. It works sort of wrapper on top of PCA954x module